apis/kubedb/v1alpha2/kafka_helpers.go

```diff
@@ -0,0 +1,318 @@
+func (k *Kafka) CertSecretVolumeName(alias KafkaCertificateAlias) string {
+	return string(alias) + "-certs"
+}
+
+// returns CertSecretVolumeMountPath
+// if configDir is "/opt/kafka/config",
+// mountPath will be, "/opt/kafka/config/<alias>".
+func (k *Kafka) CertSecretVolumeMountPath(configDir string, cert string) string {
+	return filepath.Join(configDir, cert)
+}
+
+func (k *Kafka) PVCName(alias string) string {
+	return meta_util.NameWithSuffix(k.Name, alias)
+}
+
+func (k *Kafka) SetHealthCheckerDefaults() {
+	if k.Spec.HealthChecker.PeriodSeconds == nil {
+		k.Spec.HealthChecker.PeriodSeconds = pointer.Int32P(20)
+	}
+	if k.Spec.HealthChecker.TimeoutSeconds == nil {
+		k.Spec.HealthChecker.TimeoutSeconds = pointer.Int32P(10)
+	}
+	if k.Spec.HealthChecker.FailureThreshold == nil {
+		k.Spec.HealthChecker.FailureThreshold = pointer.Int32P(3)
+	}
+}
+
+func (k *Kafka) SetDefaults() {
+	if k.Spec.TerminationPolicy == "" {
+		k.Spec.TerminationPolicy = TerminationPolicyDelete
+	}
+
+	if k.Spec.StorageType == "" {
+		k.Spec.StorageType = StorageTypeDurable
+	}
+
+	if k.Spec.Topology != nil {
+		if k.Spec.Topology.Controller != nil {
+			if k.Spec.Topology.Controller.Suffix == "" {
+				k.Spec.Topology.Controller.Suffix = string(KafkaNodeRoleController)
+			}
+			if k.Spec.Topology.Controller.Replicas == nil {
+				k.Spec.Topology.Controller.Replicas = pointer.Int32P(1)
+			}
+			apis.SetDefaultResourceLimits(&k.Spec.Topology.Controller.Resources, DefaultResources)
+		}
+
+		if k.Spec.Topology.Broker != nil {
+			if k.Spec.Topology.Broker.Suffix == "" {
+				k.Spec.Topology.Broker.Suffix = string(KafkaNodeRoleBroker)
+			}
+			if k.Spec.Topology.Broker.Replicas == nil {
+				k.Spec.Topology.Broker.Replicas = pointer.Int32P(1)
+			}
+			apis.SetDefaultResourceLimits(&k.Spec.Topology.Broker.Resources, DefaultResources)
+		}
+	} else {
+		apis.SetDefaultResourceLimits(&k.Spec.PodTemplate.Spec.Resources, DefaultResources)
+		if k.Spec.Replicas == nil {
+			k.Spec.Replicas = pointer.Int32P(1)
+		}
+	}
+	if k.Spec.EnableSSL {
+		k.SetTLSDefaults()
+	}
+	k.SetHealthCheckerDefaults()
+}
+
+func (k *Kafka) SetTLSDefaults() {
+	if k.Spec.TLS == nil || k.Spec.TLS.IssuerRef == nil {
+		return
+	}
+	k.Spec.TLS.Certificates = kmapi.SetMissingSecretNameForCertificate(k.Spec.TLS.Certificates, string(KafkaServerCert), k.CertificateName(KafkaServerCert))
+	k.Spec.TLS.Certificates = kmapi.SetMissingSecretNameForCertificate(k.Spec.TLS.Certificates, string(KafkaClientCert), k.CertificateName(KafkaClientCert))
+}
+
+// ToMap returns ClusterTopology in a Map
+func (kfTopology *KafkaClusterTopology) ToMap() map[KafkaNodeRoleType]KafkaNode {
+	topology := make(map[KafkaNodeRoleType]KafkaNode)
+
+	if kfTopology.Controller != nil {
+		topology[KafkaNodeRoleController] = *kfTopology.Controller
+	}
+	if kfTopology.Broker != nil {
+		topology[KafkaNodeRoleBroker] = *kfTopology.Broker
+	}
+	return topology
+}
+
+type KafkaApp struct {
+	*Kafka
```

Comment:
Is there any usage of these ? It not , we can remove

Reply:
Yes, they are used to get the appBindingMetadata.